### PR TITLE
Add sample-based NeuRD loss

### PR DESCRIPTION
**Pull request by godmoves**
This pull request adds the sample-based NeuRD loss with regular policy entropy. More experiment results are in #891.

### CODE REVIEW

**Comment by lanctot** [comment]
Great! Can you add a test to this file: https://github.com/deepmind/open_spiel/blob/master/open_spiel/python/pytorch/policy_gradient_pytorch_test.py

Maybe you could run it on Kuhn poker (for a low number of iterations, so the test doesn't take too long) and then check the exploitability is lower than some threshold?

**Comment by godmoves** [comment]
I added the results of NeuRD on Kuhn poker in #891.

**Comment by lanctot** [comment]
@godmoves the tests failed.. any idea why? (You can see the logs by clicking on them below)

**Comment by godmoves** [comment]
@lanctot It seems that we step all players in the Hanabi test, and this causes some `None` values to the agent's action history. An error occurs when we try to convert the action history to tensor.

This is not triggered before because the game ends in just 7 steps. 7 is small than the batch size so no training will happen. After the NeuRD loss test is added, the random seed changes and the game lasts more than 16 steps, and this causes the `None` value error mentioned above.

https://github.com/deepmind/open_spiel/blob/ae9371a1c92a6bfb2312e8315882bfe7d3e69a1a/open_spiel/python/pytorch/policy_gradient_pytorch_test.py#L102-L106

**Comment by lanctot** [comment]
I see, ok how about just having a separate test for NeuRD in that file, which runs it on Kuhn for 100 episodes and checks the exploitability is less than 1?

**Comment by lanctot** [comment]
The main purpose is to have a test in place that runs it, so that if someone else makes a change and it breaks NeuRD it will get caught within this test.

**Comment by godmoves** [comment]
@lanctot  I added a test of NeuRD on Kuhn poker. But I'm not sure if it's appropriate to use the code from the example file in my test.

The upper bound value of 0.7 is determined by running the test with 500 different seeds, and it seems 0.7 may be a good value to check potential changes that break the NeuRD code.

o2AWci5tyB

**Comment by lanctot** [comment]
Should be ok as long as the seed is fixed in the test, it will be fully deterministic.

**Comment by lanctot** [comment]
Thanks, I agree 0.7 is a good choice given your graph.
